kfs-ld/src/main/java/edu/arizona/kfs/module/ld/batch/service/impl/LaborEncumbranceAdjustmentServiceImpl.java

```diff
@@ -0,0 +1,626 @@
+     */
+    public boolean buildEncumbranceDifferenceFile(File inputFile, File balanceFile, File outputFile, File errorFile, File reconFile) {
+        accountingCycleCachingService.initialize();
+        BufferedReader inputReader = null;
+        BufferedReader balanceReader = null;
+        PrintStream outputStream = null;
+        PrintStream errorStream = null;
+        PrintStream reconStream = null;
+        Date currentDate = dateTimeService.getCurrentSqlDate();
+        LaborOriginEntry lastOriginEntry = null;
+        LaborOriginEntryToBalanceRecordComparator lineComparator = new LaborOriginEntryToBalanceRecordComparator();
+        String finOriginCode = null;
+        
+        // field length map used for parsing the balance file
+        Map<String, Integer> lMap = new LaborOriginEntryFieldUtil().getFieldLengthMap();
+
+        // counters for reporting
+        LaborEncumbranceJobReportData jobData = new LaborEncumbranceJobReportData();
+        
+        int lineNumber = 1;
+
+        try {
+            // open all the files
+            inputReader = new BufferedReader(new FileReader(inputFile));
+            balanceReader = new BufferedReader(new FileReader(balanceFile));
+            outputStream = new PrintStream(outputFile);
+            errorStream = new PrintStream(errorFile);
+
+            LOG.info("All Files Opened - starting comparisons");
+            // ok, now start up the iterators
+            String encumbranceLine = null;
+            String balanceLine = null;
+            LaborOriginEntry encumbranceEntryLine = new LaborOriginEntry();
+            KualiDecimal reconAmount = KualiDecimal.ZERO;
+
+            // get the first line of each
+            encumbranceLine = inputReader.readLine();
+            balanceLine = balanceReader.readLine();
+            
+            finOriginCode = paramSrv.getParameterValueAsString("KFS-GL", "Batch", "MANUAL_FEED_ORIGINATION"); 
+                        
+            // keep processing until *both* have been exhausted
+            // the encumbrance file is assumed to be a comprehensive list of
+            // the current balances, so any "leftovers" in the balance file will
+            // need to be removed
+            while (encumbranceLine != null || balanceLine != null) {
+                if ( LOG.isDebugEnabled() ) {
+                    LOG.debug("Processing Lines:\nHCM: "+encumbranceLine + "\nBAL: "+balanceLine);
+                }
+                boolean encumbranceLineReadError = false;
+                boolean advanceEncumbranceFile = false;
+                boolean advanceBalanceFile = false;
+                if ( encumbranceLine != null ) {
+                    try {
+                    	// first remove the invalid characters that's being sent from HCM.
+                    	encumbranceLine = encumbranceLine.replaceAll("[^\\p{ASCII}]", " ");
```

Comment:
Woof, magic string for sure -- usually I don't request a usage of a string literal to be extracted to a static final String unless it's used multiple times, but I think this is an exceptional case.
Please create a class constant for this, and perhaps a description of what the target char is that's being scrubbed. Doing so highlights that something none-standard has been employed, and will save maintainers a trip to search out the exact intended purpose (I do appreciate the short blurb already there, but think slightly more detail is warranted).